### ISSUE TITLE
0.3.0 release

### ISSUE THREAD

**Issue by cdrage** [streamlink/streamlink#467]
Been a *whileeeeeee* since another release and now we have a brand-spanking new travis builder + release script with rst instead of md.

Hopefully this time I don't build the Windows binary incorrectly (forgot to remove the old build folder last time!)

Everyone else ready for a 0.3.0 release?

@streamlink/members

**Comment by gravyboat** [comment]
Yeah I think it's time for another release. Now that most of the build itself is automated, can you note the steps you're using just so we have it for future record in the event we change something?

**Comment by cdrage** [comment]
Hey @gravyboat 

So basically I just run `./release.sh` in the scripts folder.

Unfortunately I encountered some errors this time and created a PR #480 for it.

Let's see if the Windows binary is automatically added to the new tagged release!

**Comment by cdrage** [comment]
AWESOME. Looks like it just did (got an email notification)

**Comment by gravyboat** [comment]
@cdrage Awesome! So at this point we aren't doing anything manually other than these steps:

1. Ensure pypi creds are accurate and set and you have proper github perms for creating the release.
2. Create a new tag within github.
3. Run `scripts/release.sh` in a presumably clean virtualenv.

Does that look accurate to you?

**Comment by stevek123** [comment]
Why did the size grew so much? Are there that many new code?

**Comment by mmetak** [comment]
@stevek123 I believe it's mostly due to windows ffmpeg binary (~54MB). lol ;-)

**Comment by cdrage** [comment]
@gravyboat 

Yup. That's accurate.

Requirements are listed in the release script (gpg, github-release, git, etc.) and you need the correct git + python requirements.

**Comment by mmetak** [comment]
@cdrage @gravyboat @beardypig 
Would it be possible to provide a release tarball without win32 binaries?
It's not really practical for non-win users and maintainers who build it from source. Fedora repackages the official tarball without those binaries and the difference is 29MB/0.5MB. Huge.

Anyway, I already modified the arch pkgbuild file to use Fedora source. ;-)

**Comment by beardypig** [comment]
IMO, a much more suitable thing to upload would be the `tar.gz` file produced by

```sh
python setup.py sdist
```

Which just includes the python code, as you would find on pypi.

**Comment by bastimeyer** [comment]
Is there a chance for a `3.0.1` release when this is fixed?
The `0.3.0` one has a couple of other annoying issues that have already been fixed.

**Comment by gravyboat** [comment]
Yeah I think we should do a 0.3.0 release as well, what do you think @beardypig and @cdrage?

**Comment by beardypig** [comment]
Yep, 0.3.1 would be good :-)

**Comment by cdrage** [comment]
@beardypig Fine by me.

So generate via `python setup.py sdist` and I'l upload that instead :)

Keep in mind though, that `Source Code` zip in the release page will always have the win32 zip included.

**Comment by bastimeyer** [comment]
Can we publish a new release then when the license stuff has been resolved?
#483 should be published as soon as possible, so Windows users don't have to fix their default config file.

**Comment by gravyboat** [comment]
Once https://github.com/streamlink/streamlink/pull/515 is merged let's do another release @cdrage.

**Comment by gravyboat** [comment]
https://github.com/streamlink/streamlink/pull/480 is also merged.

**Comment by cdrage** [comment]
@gravyboat Awesome! I'll create another issue for this.
